src/StatusLine.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+namespace Teapot;
+
+use Teapot\StatusCodeException\InvalidStatusCodeException;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+class StatusLine
```

Comment:
@shrikeh 

How about renaming `StatusCodeInterface` to `StatusCodeClass`? Is that even more awkward?

```php
interface StatusCodeClass
{
    const CLASS_INFORMATIONAL = 1;
    const CLASS_SUCCESSFUL = 2;
    const CLASS_REDIRECTION = 3;
    const CLASS_CLIENT_ERROR = 4;
    const CLASS_SERVER_ERROR = 5;
}
```

and then also adjusting

```php
private function is($statusCodeClass)
{
    return ($this->responseClass() === $statusCodeClass);
}
```

?